forcetree.c

```diff
@@ -329,10 +345,14 @@ int force_tree_create_nodes(const int firstnode, const int lastnode, const int n
 #endif
     for(i = 0; i < npart; i++)
     {
+        P[i].SufferFromCoupling = 0;
+
         /*Can't break from openmp for*/
         if(nnext_thread >= lastnode-1)
             continue;
 
+        const double minlen = 1.0e-3 * All.ForceSoftening[P[i].Type];
```

Comment:
Might be because it uses P[i].Type not just particle type 1?

Reply:
I think this shall really be the minimal of all particle types. Otherwise it is possible we see leaf nodes with len > minlen  and also contains multiple particles -- which breaks a major assumption of mine. But I guess your commits removes that assumption?

Reply:
Using the minimum of all particle types sounds a good idea to me.